src/Features/CSharp/Portable/Completion/CompletionProviders/ExplicitInterfaceMemberCompletionProvider.cs

```diff
@@ -82,7 +82,7 @@ internal override bool IsInsertionTrigger(SourceText text, int characterPosition
             var members = semanticModel.LookupSymbols(
                 position: name.SpanStart,
                 container: symbol)
-                    .WhereAsArray(s => !s.IsStatic)
+                    .WhereAsArray(s => !s.IsStatic && s.ContainingType.Equals(symbol))
```

Comment:
❓ Do we know why `LookupSymbols` is including these symbols in the first place? Is there a reason we don't want to implement this correction at that level instead?